lnrpc: append listchannels command with the chan_id and short_chan_id

NishantBansal2003 opened the pull request:
## Change Description
Fixes: #8650 
This PR enhances channel ID representation by introducing both integer and string formats for `ShortChannelId`:

1. **Renamed `ChanId` to `ShortChanIdInt`:**
   - Renamed to better reflect its role as the integer representation of `ShortChannelId`.

2. **Added `ShortChanId`:**
   - Introduced a string representation of `ShortChannelId` in the format `block x transaction x output`.

3. **Introduced `chan_id`:**
   - Added `chan_id` based on the definition in the [BOLT-2 Peer Protocol specification](https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#definition-of-channel_id), aligning with Lightning Network standards.

## Steps to Test
Steps for reviewers to follow to test the change.

## Pull Request Checklist
### Testing
- [x] Your PR passes all CI checks.
- [x] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [x] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [ ] [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

NishantBansal2003 commented:
PTAL @ziggie1984

ziggie1984 commented:
Please read the contribution guidelines thoroughly in particular what was immediately catching my eye is the commit structure here (https://github.com/lightningnetwork/lnd/blob/a388c1f39d849005b58eae81516b4104929101a6/docs/code_contribution_guidelines.md?plain=1#L10C8-L10C34)

NishantBansal2003 commented:
> Please read the contribution guidelines thoroughly in particular what was immediately catching my eye is the [commit structure](https://github.com/lightningnetwork/lnd/blob/a388c1f39d849005b58eae81516b4104929101a6/docs/code_contribution_guidelines.md?plain=1#L10C8-L10C34)

Got it! Working on it Now...

alexbosworth commented:
Maybe people should be warned that a renaming an API attribute would break things

ziggie1984 commented on lnrpc/lightning.proto:
```diff
@@ -1471,7 +1471,7 @@ message Channel {
     output index for the channel.
     */
-    uint64 chan_id = 4 [jstype = JS_STRING];
+    uint64 short_chan_id_int = 4 [jstype = JS_STRING];
```
we are not renaming stuff, we are deprecating it instead, so no worries @alexbosworth

ziggie1984 replied:
I think we can keep it but instead return the right chan_id not the short chan id here, but I need to think about backwards-compatibility issues.

NishantBansal2003 replied:
The right chan_id being in string format may also introduce backward compatibility issues.

NishantBansal2003 replied:
So, I can think of two solutions:  
- Mark `chan_id` as deprecated and introduce fields like `short_chan_id_int`, `short_chan_id`, and `channel_id`.  
- Introduce new names (eg.- `chan_id` for `uint64`, `chan_id_str` for string, and `channel_id` for BOLT-2 definition) for the additional fields while preserving the same name for the existing field.

wdyt?

ziggie1984 replied:
Yeah you cannot reuse this, it is uint64 and for the real channel_id we need string or bytes, so best is to start from a clean state and deprecate the old value.

NishantBansal2003 commented:
Hey @ziggie1984, I have addressed the Git commit structure and made the required changes based on your review. Please take a look.

ziggie1984 commented:
Thank you for the PR, I think it is enough to have the short_chan_id and the normal channel_id in the rpc msg. Then when for example depicting the channel struct via `lncli listchannels` we should show both representation for the short channel id, but we should not put the different types into the protofile (int&string) that should then happen at the lncli or application level.

NishantBansal2003 commented:
> Thank you for the PR, I think it is enough to have the short_chan_id and the normal channel_id in the rpc msg. Then when for example depicting the channel struct via `lncli listchannels` we should show both representation for the short channel id, but we should not put the different types into the protofile (int&string) that should then happen at the lncli or application level.

Addresses the reviewed changes. Here is how the `lncli listchannels` output now appears (on the `regtest` network):
```json
{
    "channels": [
        {
            "active": true,
            "remote_pubkey": "02769271ddfb03149bc9e2e6124bb8d83209284dda3affed0dead73f57b64f45dd",
            "channel_point": "6ab312e3b744e1b80a33a6541697df88766515c31c08e839bf11dc9fcc036a19:0",
            "chan_id": "829031767408640",
            "capacity": "20000",
            "local_balance": "16530",
            "remote_balance": "0",
            "commit_fee": "3140",
            "commit_weight": "772",
            "fee_per_kw": "2500",
            "unsettled_balance": "0",
            "total_satoshis_sent": "0",
            "total_satoshis_received": "0",
            "num_updates": "0",
            "pending_htlcs": [],
            "csv_delay": 144,
            "private": false,
            "initiator": true,
            "chan_status_flags": "ChanStatusDefault",
            "local_chan_reserve_sat": "354",
            "remote_chan_reserve_sat": "354",
            "static_remote_key": false,
            "commitment_type": "ANCHORS",
            "lifetime": "11",
            "uptime": "11",
            "close_address": "",
            "push_amount_sat": "0",
            "thaw_height": 0,
            "local_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "354",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "19800000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "remote_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "354",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "19800000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "alias_scids": [],
            "zero_conf": false,
            "zero_conf_confirmed_scid": "0",
            "peer_alias": "02769271ddfb03149bc9",
            "peer_scid_alias": "0",
            "memo": "",
            "custom_channel_data": "",
            "channel_id": "196a03cc9fdc11bf39e8081cc315657688df971654a6330ab8e144b7e312b36a",
            "short_chan_id": "829031767408640",
            "short_chan_id_str": "754x1x0"
        },
        {
            "active": true,
            "remote_pubkey": "02769271ddfb03149bc9e2e6124bb8d83209284dda3affed0dead73f57b64f45dd",
            "channel_point": "e428fb553769e4e53dbfb9749f8b057272969517ea85c63f46288487bfd62cf0:0",
            "chan_id": "807041534853120",
            "capacity": "100000",
            "local_balance": "0",
            "remote_balance": "96530",
            "commit_fee": "3140",
            "commit_weight": "772",
            "fee_per_kw": "2500",
            "unsettled_balance": "0",
            "total_satoshis_sent": "0",
            "total_satoshis_received": "0",
            "num_updates": "0",
            "pending_htlcs": [],
            "csv_delay": 144,
            "private": false,
            "initiator": false,
            "chan_status_flags": "ChanStatusDefault",
            "local_chan_reserve_sat": "1000",
            "remote_chan_reserve_sat": "1000",
            "static_remote_key": false,
            "commitment_type": "ANCHORS",
            "lifetime": "459",
            "uptime": "421",
            "close_address": "",
            "push_amount_sat": "0",
            "thaw_height": 0,
            "local_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "1000",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "99000000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "remote_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "1000",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "99000000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "alias_scids": [],
            "zero_conf": false,
            "zero_conf_confirmed_scid": "0",
            "peer_alias": "02769271ddfb03149bc9",
            "peer_scid_alias": "0",
            "memo": "",
            "custom_channel_data": "",
            "channel_id": "f02cd6bf878428463fc685ea1795967272058b9f74b9bf3de5e4693755fb28e4",
            "short_chan_id": "807041534853120",
            "short_chan_id_str": "734x1x0"
        }
    ]
}
```

ziggie1984 commented:
While looking at the protobuf file, a lot of the structs use `chan_id` for the short channel id, so I am at this point not convinced about deprecating this value but rather just adding a new field for the longer channel id to keep the change small.  

Maybe something like `funding_outpoint_id` 

I think it is not really worth dealing with all these name changes in the protofiles if so many structs are affected. Wdyt @guggero ?

ziggie1984 commented:
I have a better idea:

So initial request was to make the `lncli listchannels` output more detailed by adding the channel_id and also adding the human-readable representation of the short channel_id to the output of the command.

We can do all this by just changing the `lncli` program rather than touching the lnd core unit. The human-readable representation of the short_channel_id can be derived from the integer value currently already available. Moreover also the Xored ChannelID can be calculated using the ChannelPoint already available in the grpc msg.

So I favour the approach to not change anything in the RPC code, but just add all the information on the lncli side!

NishantBansal2003 commented:
> I have a better idea:
> 
> So initial request was to make the `lncli listchannels` output more detailed by adding the channel_id and also adding the human-readable representation of the short channel_id to the output of the command.
> 
> We can do all this by just changing the `lncli` program rather than touching the lnd core unit. The human-readable representation of the short_channel_id can be derived from the integer value currently already available. Moreover also the Xored ChannelID can be calculated using the ChannelPoint already available in the grpc msg.
> 
> So I favour the approach to not change anything in the RPC code, but just add all the information on the lncli side!

Done. Here is how the `lncli listchannels` output now appears:
```json
{
    "channels": [
        {
            "active": false,
            "remote_pubkey": "02769271ddfb03149bc9e2e6124bb8d83209284dda3affed0dead73f57b64f45dd",
            "channel_point": "6ab312e3b744e1b80a33a6541697df88766515c31c08e839bf11dc9fcc036a19:0",
            "funding_outpoint_id": "196a03cc9fdc11bf39e8081cc315657688df971654a6330ab8e144b7e312b36a",
            "chan_id": "829031767408640",
            "chan_id_str": "754x1x0",
            "capacity": "20000",
            "local_balance": "16530",
            "remote_balance": "0",
            "commit_fee": "3140",
            "commit_weight": "772",
            "fee_per_kw": "2500",
            "unsettled_balance": "0",
            "total_satoshis_sent": "0",
            "total_satoshis_received": "0",
            "num_updates": "0",
            "pending_htlcs": [],
            "csv_delay": 144,
            "private": false,
            "initiator": true,
            "chan_status_flags": "ChanStatusDefault",
            "local_chan_reserve_sat": "354",
            "remote_chan_reserve_sat": "354",
            "static_remote_key": false,
            "commitment_type": "ANCHORS",
            "lifetime": "0",
            "uptime": "0",
            "close_address": "",
            "push_amount_sat": "0",
            "thaw_height": 0,
            "local_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "354",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "19800000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "remote_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "354",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "19800000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "alias_scids": [],
            "zero_conf": false,
            "zero_conf_confirmed_scid": "0",
            "peer_alias": "02769271ddfb03149bc9",
            "peer_scid_alias": "0",
            "memo": "",
            "custom_channel_data": ""
        },
        {
            "active": false,
            "remote_pubkey": "02769271ddfb03149bc9e2e6124bb8d83209284dda3affed0dead73f57b64f45dd",
            "channel_point": "e428fb553769e4e53dbfb9749f8b057272969517ea85c63f46288487bfd62cf0:0",
            "funding_outpoint_id": "f02cd6bf878428463fc685ea1795967272058b9f74b9bf3de5e4693755fb28e4",
            "chan_id": "807041534853120",
            "chan_id_str": "734x1x0",
            "capacity": "100000",
            "local_balance": "0",
            "remote_balance": "96530",
            "commit_fee": "3140",
            "commit_weight": "772",
            "fee_per_kw": "2500",
            "unsettled_balance": "0",
            "total_satoshis_sent": "0",
            "total_satoshis_received": "0",
            "num_updates": "0",
            "pending_htlcs": [],
            "csv_delay": 144,
            "private": false,
            "initiator": false,
            "chan_status_flags": "ChanStatusDefault",
            "local_chan_reserve_sat": "1000",
            "remote_chan_reserve_sat": "1000",
            "static_remote_key": false,
            "commitment_type": "ANCHORS",
            "lifetime": "0",
            "uptime": "0",
            "close_address": "",
            "push_amount_sat": "0",
            "thaw_height": 0,
            "local_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "1000",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "99000000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "remote_constraints": {
                "csv_delay": 144,
                "chan_reserve_sat": "1000",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "99000000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "alias_scids": [],
            "zero_conf": false,
            "zero_conf_confirmed_scid": "0",
            "peer_alias": "02769271ddfb03149bc9",
            "peer_scid_alias": "0",
            "memo": "",
            "custom_channel_data": ""
        }
    ]
}
```

NishantBansal2003 commented:
> > I have a better idea:
> > So initial request was to make the `lncli listchannels` output more detailed by adding the channel_id and also adding the human-readable representation of the short channel_id to the output of the command.
> > We can do all this by just changing the `lncli` program rather than touching the lnd core unit. The human-readable representation of the short_channel_id can be derived from the integer value currently already available. Moreover also the Xored ChannelID can be calculated using the ChannelPoint already available in the grpc msg.
> > So I favour the approach to not change anything in the RPC code, but just add all the information on the lncli side!
> 
> Done. Here is how the `lncli listchannels` output now appears:
> 
> ```json
> {
>     "channels": [
>         {
>             "active": false,
>             "remote_pubkey": "02769271ddfb03149bc9e2e6124bb8d83209284dda3affed0dead73f57b64f45dd",
>             "channel_point": "6ab312e3b744e1b80a33a6541697df88766515c31c08e839bf11dc9fcc036a19:0",
>             "funding_outpoint_id": "196a03cc9fdc11bf39e8081cc315657688df971654a6330ab8e144b7e312b36a",
>             "chan_id": "829031767408640",
>             "chan_id_str": "754x1x0",
>             "capacity": "20000",
>             "local_balance": "16530",
>             "remote_balance": "0",
>             "commit_fee": "3140",
>             "commit_weight": "772",
>             "fee_per_kw": "2500",
>             "unsettled_balance": "0",
>             "total_satoshis_sent": "0",
>             "total_satoshis_received": "0",
>             "num_updates": "0",
>             "pending_htlcs": [],
>             "csv_delay": 144,
>             "private": false,
>             "initiator": true,
>             "chan_status_flags": "ChanStatusDefault",
>             "local_chan_reserve_sat": "354",
>             "remote_chan_reserve_sat": "354",
>             "static_remote_key": false,
>             "commitment_type": "ANCHORS",
>             "lifetime": "0",
>             "uptime": "0",
>             "close_address": "",
>             "push_amount_sat": "0",
>             "thaw_height": 0,
>             "local_constraints": {
>                 "csv_delay": 144,
>                 "chan_reserve_sat": "354",
>                 "dust_limit_sat": "354",
>                 "max_pending_amt_msat": "19800000",
>                 "min_htlc_msat": "1",
>                 "max_accepted_htlcs": 483
>             },
>             "remote_constraints": {
>                 "csv_delay": 144,
>                 "chan_reserve_sat": "354",
>                 "dust_limit_sat": "354",
>                 "max_pending_amt_msat": "19800000",
>                 "min_htlc_msat": "1",
>                 "max_accepted_htlcs": 483
>             },
>             "alias_scids": [],
>             "zero_conf": false,
>             "zero_conf_confirmed_scid": "0",
>             "peer_alias": "02769271ddfb03149bc9",
>             "peer_scid_alias": "0",
>             "memo": "",
>             "custom_channel_data": ""
>         },
>         {
>             "active": false,
>             "remote_pubkey": "02769271ddfb03149bc9e2e6124bb8d83209284dda3affed0dead73f57b64f45dd",
>             "channel_point": "e428fb553769e4e53dbfb9749f8b057272969517ea85c63f46288487bfd62cf0:0",
>             "funding_outpoint_id": "f02cd6bf878428463fc685ea1795967272058b9f74b9bf3de5e4693755fb28e4",
>             "chan_id": "807041534853120",
>             "chan_id_str": "734x1x0",
>             "capacity": "100000",
>             "local_balance": "0",
>             "remote_balance": "96530",
>             "commit_fee": "3140",
>             "commit_weight": "772",
>             "fee_per_kw": "2500",
>             "unsettled_balance": "0",
>             "total_satoshis_sent": "0",
>             "total_satoshis_received": "0",
>             "num_updates": "0",
>             "pending_htlcs": [],
>             "csv_delay": 144,
>             "private": false,
>             "initiator": false,
>             "chan_status_flags": "ChanStatusDefault",
>             "local_chan_reserve_sat": "1000",
>             "remote_chan_reserve_sat": "1000",
>             "static_remote_key": false,
>             "commitment_type": "ANCHORS",
>             "lifetime": "0",
>             "uptime": "0",
>             "close_address": "",
>             "push_amount_sat": "0",
>             "thaw_height": 0,
>             "local_constraints": {
>                 "csv_delay": 144,
>                 "chan_reserve_sat": "1000",
>                 "dust_limit_sat": "354",
>                 "max_pending_amt_msat": "99000000",
>                 "min_htlc_msat": "1",
>                 "max_accepted_htlcs": 483
>             },
>             "remote_constraints": {
>                 "csv_delay": 144,
>                 "chan_reserve_sat": "1000",
>                 "dust_limit_sat": "354",
>                 "max_pending_amt_msat": "99000000",
>                 "min_htlc_msat": "1",
>                 "max_accepted_htlcs": 483
>             },
>             "alias_scids": [],
>             "zero_conf": false,
>             "zero_conf_confirmed_scid": "0",
>             "peer_alias": "02769271ddfb03149bc9",
>             "peer_scid_alias": "0",
>             "memo": "",
>             "custom_channel_data": ""
>         }
>     ]
> }
> ```

Hi @ziggie1984, just following up in case my previous message was missed.

ziggie1984 commented on docs/release-notes/release-notes-0.19.0.md:
```diff
@@ -74,6 +74,12 @@
   This is a protocol gadget required for Dynamic Commitments and Splicing that
   will be added later.
 
+* [Added]((https://github.com/lightningnetwork/lnd/pull/9390)) 
```
add the release notes under the `lncli section`

ziggie1984 commented on cmd/commands/commands.go:
```diff
@@ -86,6 +95,96 @@ func replaceCustomData(jsonBytes []byte) []byte {
+		return jsonBytes
+	}
+
+	replacedBytes := chanIDPattern.ReplaceAllFunc(
+		jsonBytes, func(match []byte) []byte {
+			// Extract the numeric string from the match
+			encoded := chanIDPattern.FindStringSubmatch(
+				string(match),
+			)[1]
+
+			shortChandIDInt, err := strconv.ParseUint(encoded, 10,
+				64)
+			if err != nil {
+				return match
+			}
+
+			// Format a new JSON field for the short_chan_id with
+			// both its numeric representation and its string
+			// representation (short_chan_id_str)
+			updatedField := fmt.Sprintf(`"chan_id": "%s",
+                         "chan_id_str": "%s"`,
+				encoded,
+				lnwire.NewShortChanIDFromInt(shortChandIDInt).
+					String())
+
+			// Replace the entire match with the new structure.
+			return []byte(updatedField)
+		},
+	)
```
```suggestion
replacedBytes := chanIDPattern.ReplaceAllFunc(
		jsonBytes, func(match []byte) []byte {
			// Extract the captured scid group from the match.
			encoded := chanIDPattern.FindStringSubmatch(
				string(match),
			)[1]

			scid, err := strconv.ParseUint(encoded, 10, 64)
			if err != nil {
				return match
			}

			// Format a new JSON field for the short_chan_id with
			// both its numeric representation and its string
			// representation (short_chan_id_str).
			scidStr := lnwire.NewShortChanIDFromInt(scid).String()
			updatedField := fmt.Sprintf(
				`"scid": "%d", "scid_str": "%s"`, scid, scidStr,
			)

			// Replace the entire match with the new structure.
			return []byte(updatedField)
		},
	)
```

ziggie1984 commented on cmd/commands/commands.go:
```diff
@@ -86,6 +95,96 @@ func replaceCustomData(jsonBytes []byte) []byte {
+		return jsonBytes
+	}
+
+	replacedBytes := channelPointPattern.ReplaceAllFunc(
+		jsonBytes, func(match []byte) []byte {
+			channelPoint := channelPointPattern.FindStringSubmatch(
+				string(match),
+			)[1]
+
+			channelOutPoint, err := wire.NewOutPointFromString(
+				channelPoint)
+			if err != nil {
+				return match
+			}
+
+			// Format a new JSON field computed from the
+			// channel_point (funding_outpoint_id).
+			updatedField := fmt.Sprintf(`"channel_point": "%s",
+                         "funding_outpoint_id": "%s"`,
+				channelPoint,
+				lnwire.NewChanIDFromOutPoint(*channelOutPoint).
+					String())
+
+			// Replace the entire match with the new structure.
+			return []byte(updatedField)
+		},
+	)
```
```suggestion
	replacedBytes := channelPointPattern.ReplaceAllFunc(
		jsonBytes, func(match []byte) []byte {
			chanPoint := channelPointPattern.FindStringSubmatch(
				string(match),
			)[1]

			chanOutpoint, err := wire.NewOutPointFromString(
				chanPoint)
			if err != nil {
				return match
			}

			// Format a new JSON field computed from the
			// channel_point (funding_outpoint_id).
			chanID := lnwire.NewChanIDFromOutPoint(*chanOutpoint)
			updatedField := fmt.Sprintf(
				`"chan_point": "%s", "chan_id": "%s"`,
				channelPoint, chanID.String(),
			)

			// Replace the entire match with the new structure.
			return []byte(updatedField)
		},
	)
```

ziggie1984 commented on cmd/commands/commands.go:
```diff
@@ -86,6 +95,96 @@ func replaceCustomData(jsonBytes []byte) []byte {
+			// with chan_point.
+			chanID := lnwire.NewChanIDFromOutPoint(*chanOutpoint)
+			updatedField := fmt.Sprintf(
+				`"chan_point": "%s", "chan_id": "%s"`,
```
I would say let's keep the channel_point to not make things too confusing.

ziggie1984 commented on cmd/commands/commands.go:
```diff
@@ -122,6 +221,10 @@ func printRespJSON(resp proto.Message) {
 
 	jsonBytesReplaced := replaceCustomData(jsonBytes)
 
+	jsonBytesReplaced = replaceChannelPoint(jsonBytesReplaced)
```
it is not necessarily a replacement but also an amendment can you add that fact in the comment.

ziggie1984 commented on cmd/commands/commands.go:
```diff
@@ -122,6 +221,10 @@ func printRespJSON(resp proto.Message) {
 
+	jsonBytesReplaced = replaceChannelPoint(jsonBytesReplaced)
+
+	jsonBytesReplaced = replaceChanID(jsonBytesReplaced)
```
same here

ziggie1984 commented on docs/release-notes/release-notes-0.19.0.md:
```diff
@@ -111,6 +111,13 @@
   now update the channel policy if the edge was not found in the graph
   database if the `create_missing_edge` flag is set.
 
+* [Added](https://github.com/lightningnetwork/lnd/pull/9390) `scid_str`, a 
+  string representation of `chan_id` in the format 
+  (`block x transaction x output`). This update replaces `chan_id` with `scid`, 
+  `channel_point` with `chan_point`, and includes `chan_id` to align with the 
+  BOLT-2 Peer Protocol specification for standardized channel identification in 
+  the `lncli listchannels` output.
```
```suggestion
[Enhance](https://github.com/lightningnetwork/lnd/pull/9390) the `lncli listchannels` output by adding the human readable short channel id and the channel id defined in BOLT02. Moreover change the misnomer of `chan_id` which was describing the short channel id to `scid` to represent what it really is.
```

ziggie1984 reviewed (changes requested):
Great work, had some final comments.

ziggie1984 reviewed (approved):
LGTM 🎉

guggero reviewed (commented):
Thanks for the PR. Approach ACK.

NishantBansal2003 commented on lnwire/short_channel_id.go:
```diff
@@ -53,7 +53,7 @@ func (c ShortChannelID) ToUint64() uint64 {
 // String generates a human-readable representation of the channel ID.
 func (c ShortChannelID) String() string {
-	return fmt.Sprintf("%d:%d:%d", c.BlockHeight, c.TxIndex, c.TxPosition)
+	return fmt.Sprintf("%dx%dx%d", c.BlockHeight, c.TxIndex, c.TxPosition)
```
Since using 'x' as a separator is also mentioned in [the BOLTs documentation](https://github.com/lightning/bolts/blob/master/07-routing-gossip.md#definition-of-short_channel_id), I believe adding a new method `AltString()` would be better for future use.

ziggie1984 reviewed (changes requested):
Can you sqash the commits

guggero commented on lnwire/short_channel_id.go:
```diff
@@ -56,6 +56,12 @@ func (c ShortChannelID) String() string {
 	return fmt.Sprintf("%d:%d:%d", c.BlockHeight, c.TxIndex, c.TxPosition)
 }
 
+// String generates a human-readable representation of the channel ID
```
The Godoc domment needs to match the actual method name here.

guggero commented on cmd/commands/commands.go:
```diff
@@ -86,6 +95,95 @@ func replaceCustomData(jsonBytes []byte) []byte {
+			)[1]
+
+			chanOutpoint, err := wire.NewOutPointFromString(
+				chanPoint)
```
nit: closing parenthesis should go to a new line.

guggero reviewed (commented):
Nice, thanks for the update. Two more nits then we're good to go!

ziggie1984 commented:
Maybe I was not clear enough in my previous comment but I meant to squash unnecessary commits that didn't mean squashing the release notes for example.

ziggie1984 commented on cmd/commands/commands_test.go:
```diff
@@ -127,10 +127,9 @@ func TestReplaceCustomData(t *testing.T) {
 	testCases := []struct {
-		name        string
-		data        string
-		replaceData string
```
was this an unused variable ?

NishantBansal2003 replied:
yes

ziggie1984 replied:
weird that golang did not notice that, did your IDE notice that ?

NishantBansal2003 replied:
No, actually, I only noticed when I copied the struct.

ziggie1984 commented on cmd/commands/commands_test.go:
```diff
@@ -191,3 +190,139 @@ func TestReplaceCustomData(t *testing.T) {
+		},
+		{
+			name: "valid json with replacement",
+			data: "{\"foo\":\"bar\",\"chan_id\":\"" +
```
Nit: I think when creating json strings it is always good to use ` instead of escaping all those special characters. For future changes.

ziggie1984 reviewed (approved):
Can squash the unit test commit with the commit which introduces the functions.

LGTM
